src/utils/reductions.hpp

```diff
@@ -70,16 +70,26 @@ int GetSize(U &v) {
 
 #ifdef MPI_PARALLEL
 // Generic for containers including Kokkos::View and std::vector
-// strips the pointers off of the type
-template <template <typename U, typename... Args> class Container, typename U,
-          typename... Args>
-MPI_Datatype GetType(Container<U, Args...> &v) {
-  return MPITypeMap<typename cleantypes::remove_all_pointers<U>::type>::type();
```

Comment:
If `remove_all_pointers` this is gone, can we then also remove `cleantypes.hpp` entirely? As far as I can tell this was the only place it was used.

Reply:
It looks like it is the only place it is used, but I don't see any reason to get rid of it though.